pkg/services/ngalert/api/api_alertmanager_test.go

```diff
@@ -138,3 +140,107 @@ func TestStatusForTestReceivers(t *testing.T) {
 		}}))
 	})
 }
+
+func TestAlertmanagerConfig(t *testing.T) {
+	sut := createSut(t)
+
+	t.Run("assert 404 Not Found when applying config to nonexistent org", func(t *testing.T) {
+		rc := models.ReqContext{
+			SignedInUser: &models.SignedInUser{
+				OrgRole: models.ROLE_EDITOR,
+				OrgId:   5,
+			},
+		}
+		request := apimodels.PostableUserConfig{}
+
+		response := sut.RoutePostAlertingConfig(&rc, request)
+
+		require.Equal(t, 404, response.Status())
+		require.Contains(t, string(response.Body()), "Alertmanager does not exist for this organization")
+	})
+
+	t.Run("assert 403 Forbidden when applying config while not Editor", func(t *testing.T) {
+		rc := models.ReqContext{
+			SignedInUser: &models.SignedInUser{
+				OrgRole: models.ROLE_VIEWER,
+				OrgId:   1,
+			},
+		}
+		request := apimodels.PostableUserConfig{}
+
+		response := sut.RoutePostAlertingConfig(&rc, request)
+
+		require.Equal(t, 403, response.Status())
+		require.Contains(t, string(response.Body()), "permission denied")
+	})
+
+	t.Run("assert 202 when config successfully applied", func(t *testing.T) {
+		rc := models.ReqContext{
+			SignedInUser: &models.SignedInUser{
+				OrgRole: models.ROLE_EDITOR,
+				OrgId:   1,
+			},
+		}
+		request := apimodels.PostableUserConfig{}
+
+		response := sut.RoutePostAlertingConfig(&rc, request)
+
+		require.Equal(t, 202, response.Status())
+	})
+
+	t.Run("assert 202 when alertmanager to configure is not ready", func(t *testing.T) {
```

Comment:
This is the test that actually exercises the bug in question. The others were auxiliary, and helped me build out the test infrastructure.